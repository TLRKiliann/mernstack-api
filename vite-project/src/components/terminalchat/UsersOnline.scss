/*user online side*/
.user--online {
  width: 25%;
  display: flex;
  flex-direction: column;
  font-size: 0.8rem;
  font-weight: bold;
  background-color: rgba(10,10,100, 0.9);
  border-left: 1px solid slategray;
  border-radius: 0px 15px 15px 0px;
  color: #50bcfe;
  overflow: scroll;
  overflow-x: hidden;
  
  .div--userolinetitle {
    background-color: rgba(20,10,100, 0.9);
    border-bottom: 1px dashed slategray;
    border-radius: 0px 15px 0px 0px;
  }
  .userolinetitle {
    margin-left: 20px;
    font-family: "Junction", sans-serif;
    color: slategray;
  }
}

.all--usersbanner {
  margin-left: 20px;
  width: 100%;
  margin-top: 10px;
  display: flex;
  flex-direction: row;
  border-bottom: 1px dashed darkslategrey;

  .smile--img {
    margin-right: 10px;
    border-radius: 50% 50%;
    object-fit: cover;
  }
}

.para--spanchat {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;

  .span--useronline {
    margin-left: 10px;
    font-size: 20px;
    text-decoration: none;
  }
  .styleusr--span:hover {
    cursor: pointer;
  }
}