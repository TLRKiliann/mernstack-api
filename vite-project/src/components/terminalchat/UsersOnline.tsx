import React, { useState } from 'react'
import { Link } from "react-router-dom"
import { UserType } from '../models/usertype'
import './UsersOnline.scss'

interface UserOnlineProps {
  user: UserType
  roomStyle: object
  handleAskUserPrivate: (order_id: number) => void
}

const UserOnline: React.FC = (props: UserOnlineProps) => {
  
  return(
    props.user.room === props.roomStyle && (
      <div key={props.user.order_id} className="all--usersbanner">   
        <div className="username--useronline">
          <img
            src={props.user.img}
            width="100%"
            height="100%"
            className="smile--img"
            alt={props.user.img}
          /> 

          <span className="useronline--username">
            {props.user.firstName}
          </span>

        </div>
        
        <p className="para--spanchat"> 
          {props.user.isConnected ? (
            <span
              className="span--useronline connector--icon"
              style={{color: 'lightgreen'}}
            >
              ✔
            </span>
            ) : (
            <span 
              className="span--useronline connector--icon"
            >
              ❌
            </span>
            )
          }

          <span 
            onClick={() => props.handleAskUserPrivate(props.user.order_id)}
            className="span--useronline styleusr--span"
            style={{color: 'orange', fontSize: '22px'}}
          >
            ✉
          </span>

        </p>
      </div>
    )
  )
}

export default UserOnline
