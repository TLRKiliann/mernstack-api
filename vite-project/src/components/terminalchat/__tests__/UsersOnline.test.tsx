import React from "react"
import { screen, render, fireEvent } from '@testing-library/react'
import { create } from "react-test-renderer"
import { expect, vi } from 'vitest'
import '@testing-library/jest-dom'
import UsersOnline from "../UsersOnline.tsx"
import { handleAskUserPrivate } from '../UsersOnline.tsx'

test('MatchSnapShot test UsersOnline', () => {
  const tree = create(<UsersOnline />)
  expect(tree.toJSON()).toMatchSnapshot();
});

describe("testing children UsersOnline", () => {
  const Span = ({onClick, children}) => (
    <span onClick={onClick} data-testid="span-useronline">
      {children}
    </span>
  )
  test('calls onClick prop when clicked ✉', () => {
  const todo = { id: 2 }
  const handleAskUserPrivate = vi.fn(todo => todo.id)
  handleAskUserPrivate(todo)
  render(
    <span
      data-testid="span-useronline"
      onClick={handleAskUserPrivate}>
      ✉
    </span>
  )
  fireEvent.click(screen.getByText(/✉/i))
  expect(handleAskUserPrivate).toHaveReturnedWith(2)
  expect(handleAskUserPrivate).toHaveBeenCalledTimes(2)
  })
});
