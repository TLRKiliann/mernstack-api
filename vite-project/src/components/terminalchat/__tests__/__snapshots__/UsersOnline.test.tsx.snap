// Vitest Snapshot v1

exports[`MatchSnapShot test UsersOnline 1`] = `
<div
  className="all--usersbanner"
>
  <div
    className="username--useronline"
  >
    <img
      className="smile--img"
      height="100%"
      width="100%"
    />
    <span
      className="useronline--username"
    />
  </div>
  <p
    className="para--spanchat"
  >
    <span
      className="span--useronline connector--icon"
    >
      ❌
    </span>
    <span
      className="span--useronline styleusr--span"
      data-testid="span-useronline"
      onClick={[Function]}
      style={
        {
          "color": "sandybrown",
          "fontSize": "22px",
        }
      }
    >
      ✉
    </span>
  </p>
</div>
`;
